projects/packages/forms/src/contact-form/class-contact-form-field.php

```diff
@@ -817,19 +817,75 @@ private function render_file_field( $id, $label, $class, $required, $required_fi
 			$input_attributes .= $attr . ( $value !== '' ? '="' . $value . '"' : '' ) . ' ';
 		}
 
-		$field  = $this->render_label( 'file', $id, $label, $required, $required_field_text );
-		$field .= "<div class='jetpack-form-file-field__dropzone' data-id='" . esc_attr( $id ) . "'>\n";
-		$field .= "<a href='#' class='wp-block-button__link wp-element-button'>" . esc_html__( 'Select a file', 'jetpack-forms' ) . "</a>\n";
-		$field .= "<span class='jetpack-form-file-field__short'>" . esc_html__( '....or drag and drop a file.', 'jetpack-forms' ) . "</span>\n";
-		$field .= '<input ' . trim( $input_attributes ) . "/>\n";
-		$field .= "<input type='hidden' name='" . esc_attr( $id ) . "_token' class='jetpack-form-file-field__token' value='' />\n";
-		$field .= "<div class='jetpack-form-file-field__preview-wrap' aria-live='polite'></div>\n";
-		$field .= "</div>\n";
+		$max_file_size = wp_max_upload_size();
+		$file_size_units = array(
+			_x( 'B', 'unit symbol', 'jetpack-forms' ),
+			_x( 'KB', 'unit symbol', 'jetpack-forms' ),
+			_x( 'MB', 'unit symbol', 'jetpack-forms' ),
+			_x( 'GB', 'unit symbol', 'jetpack-forms' ),
+		);
+
+		$global_state = array(
+			'i18n'           => array(
+				'language'           => get_bloginfo( 'language' ),
+				'fileSizeUnits'      => $file_size_units,
+				'removeFile'         => __( 'Remove', 'jetpack-forms' ),
+				'uploadError'        => __( 'Error uploading file', 'jetpack-forms' ),
+				'folderNotSupported' => __( 'Folder uploads are not supported', 'jetpack-forms' ),
+				// translators: %s is the formatted maximum file size.
+				'fileTooLarge'       => sprintf( __( 'File is too large. Maximum allowed size is %s.', 'jetpack-forms' ), size_format( $max_file_size ) ),
+				'invalidType'        => __( 'This file type is not allowed', 'jetpack-forms' ),
+			),
+			'maxUploadSize'  => $max_file_size,
+			'endpoint' => rest_url( 'wpcom/v2/unauth-file-upload' ),
+			'wp_nonce'       => wp_create_nonce( 'wp_rest' ),
+			'jp_nonce'       => wp_create_nonce( 'jetpack_file_upload_jetpack-form' ),
```

Comment:
In fact, you can use config to pass any value that doesn't need to be dynamic/reactive on the frontend.

Reply:
These have also been handled via config now.